src/salt_test/__init__.py

```diff
@@ -124,24 +165,50 @@ def _list_files(pkg: str) -> typing.List[str]:
     return root
 
 
-def testsuite_root_to_bindir(root: str) -> str:
-    """Compute /bin from the testsuite root.
+def flavor_pytest_cmd(flavor: str) -> str:
+    """Compute pytest command for a given Python flavor.
 
-    This is needed to use the correct pytest executable.
+    Different flavors can be co-installed, we should use the same Python flavor
+    for running `pytest` that is being tested.
+    """
+    if flavor == "bundle":
+        return "/usr/lib/venv-salt-minion/bin/pytest"
+    else:
+        match = re.match(r"python(\d+)$", flavor)
+        if match is not None:
+            ver = match.groups()[0]
+            if len(ver) == 1:
+                return "/usr/bin/pytest"
```

Comment:
I noticed an issue when trying to run the tests with Python 3.6 in SLE15SP4, it always runs tests with Python 3.11.

IIUC this should fix the problem:
```suggestion
                return "/usr/bin/pytest-3.6"
```

Reply:
I'd prefer to stick to the version-less `pytest` binary when the flavor is `python3` without a minor version. I need to check why `/usr/bin/pytest` picks the wrong flavor on SP4.

Reply:
Can you share how to reproduce this? I don't see any issue running `/usr/bin/pytest` on my test system. Do you maybe have `update-alternatives` configured for `pytest`?